src/core/geometry/qgsregularpolygon.h

```diff
@@ -0,0 +1,213 @@
+     */
+    QgsRegularPolygon( const QgsPointV2 center, const QgsPointV2 pt1, const int  numSides, const int circle );
+
+    /** Constructs a regular polygon by two points of the first side.
+     * \param pt1 The first vertice of the first side, also first vertice of the regular polygon.
+     * \param pt2 The second vertice of the first side.
+     * \param numSides Number of sides of the regular polygon.
+     */
+    QgsRegularPolygon( const QgsPointV2 pt1, const QgsPointV2 pt2, const int  numSides );
+
+    virtual bool operator ==( const QgsRegularPolygon &rp ) const;
+    virtual bool operator !=( const QgsRegularPolygon &rp ) const;
+
+    //! A regular polygon is empty if radius equal to 0 or number of sides < 3
+    virtual bool isEmpty() const;
+
+    /** Returns the center point of the regular polygon.
+     * \see setCenter()
+     */
+    QgsPointV2 center() const {return mCenter; }
+
+    /** Returns the radius.
+     * This is also the radius of the circumscribing circle.
+     * \see apothem()
+     * \see setRadius()
+     */
+    double radius() const {return mRadius; }
+
+    /** Returns the first vertice (corner) of the regular polygon.
+     * \see setVertice()
+     */
+    QgsPointV2 vertice() const {return mVertice; }
+
+    /** Returns the apothem of the regular polygon.
+     * The apothem is the radius of the inscribed circle.
+     * \see radius()
+     */
+    double apothem() const {return mRadius * cos( M_PI / mNumSides ); }
+
+    /** Returns the number of sides of the regular polygon.
+     * \see setNumSides()
+     */
+    int numSides() const {return mNumSides; }
+
+    /** Sets the center point.
+     * Radius is unchanged. The first vertice is reprojected from the new center.
+     * \see center()
+     */
+    void setCenter( const QgsPointV2 center );
+
+    /** Sets the radius.
+     * Center is unchanged. The first vertice is reprojected from the center with the new radius.
+     * \see radius()
+     */
+    void setRadius( const double radius );
+
+    /** Sets the first vertice.
+     * Radius is unchanged. The center is reprojected from the new first vertice.
+     * \see vertice()
+     */
+    void setVertice( const QgsPointV2 vertice );
+
+    /** Sets the number of sides.
+     * If numSides < 3, the number of sides is unchanged.
+     * \see numSides()
+     */
+    void setNumSides( const int numSides );
+
+    /** Returns a list of points into \a pts.
+     * \param pts List of points returned.
+     */
+    void points( QgsPointSequence &pts SIP_OUT ) const;
+
+    /** Returns as a polygon.
+     */
+    QgsPolygonV2 *toPolygon( ) const SIP_FACTORY;
+
+    /** Returns as a linestring.
+     */
+    QgsLineString *toLineString( ) const SIP_FACTORY;
+
+    /** Returns as a triangle.
+     * An empty triangle is returned if the regular polygon is empty.
+     */
+    QgsTriangle toTriangle( ) const;
```

Comment:
I will rewrite the comment by including the empty return if the number of sides is less than 3